SMatrixADT.c

```diff
@@ -72,12 +72,12 @@ int main ()
                 printf ("Matrix SUBTRACTION error!\n");
             break;
         case TRANSPOSE:
-            if (SMTX_transpose (&A, &B, &C)!= SMTX_ERROR)
-                SMTX_print (&C);
+            if (SMTX_transpose (&A, &B)!= SMTX_ERROR)
+                SMTX_print (&B);
             else
                 printf ("Matrix TRNSPOSE error!\n");
-
-        case Multiply:
+            break;
```

Comment:
Missing break statement in the TRANSPOSE case, causing fall-through to the next case and potential unintended behavior.